diff --git a/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml b/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
index eeb969a..917c751 100644
--- a/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
+++ b/usr/share/lipstick-jolla-home-qt5/lockscreen/LockItem.qml
@@ -225,7 +225,7 @@ SilicaFlickable {
             ]
         }
 
-        WeatherIndicatorLoader {
+        /*WeatherIndicatorLoader {
             anchors {
                 top: clock.bottom
                 horizontalCenter: clock.horizontalCenter
@@ -233,7 +233,7 @@ SilicaFlickable {
             opacity: clock.opacity
             temperatureFontPixelSize: clock.weekdayFont.pixelSize
             active: visible
-        }
+        }*/
 
         MprisPlayerControls {
             id: mpris
