diff --git a/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutLandscape.qml b/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutLandscape.qml
index a03f396..1302818 100644
--- a/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutLandscape.qml
+++ b/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutLandscape.qml
@@ -104,8 +104,8 @@ KeyboardLayout {
         }
         CharacterKey {
             key: Qt.Key_Multi_key
-            text: "*p"
-            caption: "*p"
+            text: " *p"
+            caption: "␣*p"
             height: geometry.keyHeightPortrait
             width: main.width / 10
             showPopper: false
diff --git a/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutPortrait.qml b/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutPortrait.qml
index e617a31..6571929 100644
--- a/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutPortrait.qml
+++ b/usr/share/maliit/plugins/com/jolla/PhoneNumberLayoutPortrait.qml
@@ -71,8 +71,8 @@ KeyboardLayout {
         }
         CharacterKey {
             key: Qt.Key_Multi_key
-            text: "*p"
-            caption: "*p"
+            text: " *p"
+            caption: "␣*p"
             height: geometry.keyHeightPortrait
             width: main.width / 4
             showPopper: false
