diff --git a/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml b/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
index 1577c1b..251b5ff 100644
--- a/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
+++ b/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
@@ -33,16 +33,23 @@ import com.jolla.keyboard 1.0
 KeyboardRow {
     splitIndex: 3
 
-    SymbolKey {}
-    ContextAwareCommaKey {}
+    SymbolKey { implicitWidth: punctuationKeyWidth }
+    ContextAwareCommaKey { accents: "@/,;"; accentsShifted: accents; }
+    CharacterKey {
+        caption:        "'"; accents: "\"#„“”«»*"
+        captionShifted: "'"; accentsShifted: accents
+        implicitWidth: punctuationKeyWidth
+        fixedWidth: !splitActive
+        separator: SeparatorState.HiddenSeparator
+    }
     SpacebarKey {}
     SpacebarKey {
         active: splitActive
         languageLabel: ""
     }
     CharacterKey {
-        caption: "."
-        captionShifted: "."
+        caption: "."; accents: "×÷…·¶"
+        captionShifted: "."; accentsShifted: accents
         implicitWidth: punctuationKeyWidth
         fixedWidth: !splitActive
         separator: SeparatorState.HiddenSeparator
