diff --git a/usr/share/maliit/plugins/com/jolla/PeriodKey.qml b/usr/share/maliit/plugins/com/jolla/PeriodKey.qml
index fc78404..7cf1201 100644
--- a/usr/share/maliit/plugins/com/jolla/PeriodKey.qml
+++ b/usr/share/maliit/plugins/com/jolla/PeriodKey.qml
@@ -5,8 +5,8 @@ CharacterKey {
     property bool popupAlways: true // also in symbol view state
     caption: "."
     captionShifted: "."
-    accents: "!.?"
-    accentsShifted: "!.?"
+    accents: "!.·?…×÷¶"
+    accentsShifted: accents
     implicitWidth: punctuationKeyWidth
     fixedWidth: !splitActive
     separator: SeparatorState.HiddenSeparator
diff --git a/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml b/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
index 41a2a3b..3e851ab 100644
--- a/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
+++ b/usr/share/maliit/plugins/com/jolla/SpacebarRow.qml
@@ -33,8 +33,16 @@ import com.jolla.keyboard 1.0
 KeyboardRow {
     splitIndex: 3
 
-    SymbolKey {}
-    ContextAwareCommaKey {}
+    SymbolKey { implicitWidth: punctuationKeyWidth * 1.5 }
+    ContextAwareCommaKey {
+        accents: "@/,;"
+        accentsShifted: accents
+    }
+    CharacterKey {
+        caption:        "'"; accents: "\"#„“”«»*"
+        captionShifted: "'"; accentsShifted: accents
+        separator: SeparatorState.HiddenSeparator
+    }
     SpacebarKey {}
     SpacebarKey {
         active: splitActive
