diff -Naur /usr/share/maliit/plugins/com/jolla/KeyboardBase.qml
--- /usr/share/maliit/plugins/com/jolla/KeyboardBase.qml	2021-10-15 13:18:03.545697852 +0200
+++ /usr/share/maliit/plugins/com/jolla/KeyboardBase.qml	2021-10-27 19:20:31.397447436 +0200
@@ -71,6 +71,8 @@
     property bool closeSwipeActive
     property int closeSwipeThreshold: Math.max(currentLayoutHeight*.3, Theme.itemSizeSmall)
 
+    property real longphoneoffset: Screen.height * 0.12
+
     readonly property real currentLayoutHeight: layout ? layout.height : 2 * Theme.itemSizeHuge
     readonly property real minimumLayoutHeight: {
         var height = currentLayoutHeight
@@ -113,7 +115,7 @@
         readonly property bool current: status === Loader.Ready && PagedView.isCurrentItem
 
         width: keyboard.width
-        height: status === Loader.Error ? Theme.itemSizeHuge : implicitHeight
+        height: status === Loader.Error ? Theme.itemSizeHuge : (portraitMode ? implicitHeight + longphoneoffset : implicitHeight)
 
         source: keyboard.sourceDirectory + model.file
 
